pkg/controller/oneagent/envutils_test.go

```diff
@@ -0,0 +1,199 @@
+				ListKind: "OneAgentList",
+				Plural:   "oneagents",
+				Singular: "oneagent",
+			},
+			Scope: apiextensionsv1beta1.NamespaceScoped,
+			Subresources: &apiextensionsv1beta1.CustomResourceSubresources{
+				Status: &apiextensionsv1beta1.CustomResourceSubresourceStatus{},
+			},
+		},
+	},
+	&apiextensionsv1beta1.CustomResourceDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "virtualservices.networking.istio.io",
+		},
+		Spec: apiextensionsv1beta1.CustomResourceDefinitionSpec{
+			Group:   "networking.istio.io",
+			Version: "v1alpha3",
+			Names: apiextensionsv1beta1.CustomResourceDefinitionNames{
+				Kind:     "VirtualService",
+				ListKind: "VirtualServiceList",
+				Plural:   "virtualservices",
+				Singular: "virtualservice",
+			},
+			Scope: apiextensionsv1beta1.NamespaceScoped,
+			Subresources: &apiextensionsv1beta1.CustomResourceSubresources{
+				Status: &apiextensionsv1beta1.CustomResourceSubresourceStatus{},
+			},
+		},
+	},
+	&apiextensionsv1beta1.CustomResourceDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "serviceentries.networking.istio.io",
+		},
+		Spec: apiextensionsv1beta1.CustomResourceDefinitionSpec{
+			Group:   "networking.istio.io",
+			Version: "v1alpha3",
+			Names: apiextensionsv1beta1.CustomResourceDefinitionNames{
+				Kind:     "ServiceEntry",
+				ListKind: "ServiceEntryList",
+				Plural:   "serviceentries",
+				Singular: "serviceentry",
+			},
+			Scope: apiextensionsv1beta1.NamespaceScoped,
+			Subresources: &apiextensionsv1beta1.CustomResourceSubresources{
+				Status: &apiextensionsv1beta1.CustomResourceSubresourceStatus{},
+			},
+		},
+	},
+}
+
+func init() {
+	// Register OneAgent and Istio object schemas.
+	apis.AddToScheme(scheme.Scheme)
+}
+
+type ControllerTestEnvironment struct {
+	CommunicationHosts []string
+	Client             client.Client
+	Reconciler         *ReconcileOneAgent
+
+	server *envtest.Environment
+}
+
+func newTestEnvironment() (*ControllerTestEnvironment, error) {
+	svr := &envtest.Environment{
+		KubeAPIServerFlags: append(envtest.DefaultKubeAPIServerFlags, "--allow-privileged"),
+		CRDs:               testEnvironmentCRDs,
+	}
+
+	// TODO: we shouldn't need to set environment variables. Remove usages from our code.
+	os.Setenv(k8sutil.WatchNamespaceEnvVar, "dynatrace")
+
+	cfg, err := svr.Start()
+	if err != nil {
+		return nil, err
+	}
+
+	c, err := client.New(cfg, client.Options{Scheme: scheme.Scheme})
+	if err != nil {
+		svr.Stop()
+		return nil, err
+	}
+
+	if err = c.Create(context.TODO(), &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "token-test",
+			Namespace: DefaultTestNamespace,
+		},
+		Type: corev1.SecretTypeOpaque,
+		Data: map[string][]byte{
+			"paasToken": []byte("42"),
+			"apiToken":  []byte("43"),
+		},
+	}); err != nil {
+		svr.Stop()
+		return nil, err
+	}
+
+	e := &ControllerTestEnvironment{
+		server: svr,
+		Client: c,
+		CommunicationHosts: []string{
+			"https://endpoint1.dev.ruxitlabs.com/communication",
```

Comment:
Please get rid of `ruxitlabs` and use `dynatracelabs` instead.

Reply:
The value itself is not really relevant, changed to `https://endpoint1.test.com/communication`